CMakeLists.txt

```diff
@@ -264,7 +264,9 @@ if (SANITIZER)
     message(STATUS "The TileDB library is compiled with sanitizer ${SANITIZER} enabled")
   endif()
   add_compile_options(-DTILEDB_SANITIZER=${SANITIZER})
```

Comment:
> At some point after this PR, we need to investigate how to support the other sanitizers cleanly.

One idea to explore at a later time is to generate a synthetic triplet file at configure time.

Reply:
One thing we might do is only to support sanitizers if the triplet is predefined and not derived from a script. "Predefined" could mean any of these:
* -D on the command line
* Present as a CMake cache variable
* environment variable (maybe)
Definition on the command line would suffice for CI. Using the CMake cache would allow developers an incremental method of configuration.

Reply:
Or we can add CMake presets for ASAN and the rest of the sanitizers we might support.